API-GUIDELINES.md

```diff
@@ -1,93 +1,83 @@
-* **Consistency** We want to be consistent when offering 64 bit versions of our
-  layers, so at least within a given “group” of layers all layers should
-  either have 64bits or not.
-* **Geospatial** 64 bits is most valuable for geospatial layers, other layers
-  that are primarily designed for non-geo use cases probably do not need
-  in excess of 1.000.000x zoom.
+  64-bit support of the layers, if offered, will be through a fp64 prop settable by
+  deck.gl's users. 64-bit support is usually provided if the layers are
+  expected to be used to visualize data with extremely high dynamic range, such as
+  many cases in geospatial data visualization.
 * **2D vs 2.5D (extrusion)/3D Support**
   Simple “3D”: A “2D” layer can offer “3D” support without additional
   consideration if the cost in terms of code complexity and performance
   is small. E.g. the line layer can add a z coordinate to its positions
   with very little complexity and performance impact.
-* **Consistency within Layer Groups** - The layers withing a group should have a
-  similar ambition in terms of 3D, e.g. the core geospatial layers in general
-  try to provide 2.5D (extrusion) and low cost Z-coordinate support.
-* **Lighting and effects: TBD**
-* **Splitting Layers** - If the code is significantly different between 2D and 3D
-  it can be better to split into two layers - perhaps even into two layer
-  groups. Some effort should be done to clean up code and check if sharing
-  of some parts is possible.
+* **Lighting and effects**
+  Layers with extrusion support (2.5D) or 3D should repsond correctly to lights through
+  the lighting packages provided by deck.gl. This includes providing necessary APIs for
+  setting light parameters.
+* **Splitting Layers** - If the code is significantly different different variant of the
+  same layer (such as 2D and 3D), splitting into two might be necessary - perhaps even
+  into two layer groups. Some effort should be done to clean up code and check if code
+  reuse is possible.
 * **Functional Variants of Layers**
-  Examples are Brushed, Animated, Enhanced etc layers.
-  Unless of a compelling reason, we will probably want to provide these as
-  examples rather than supported layers. When we have dynamic shaders we
-  may be able to auto inject such features into existing layers using a
-  handful of props.
-
+  Functional variants of layers such are Brushed, Animated, Enhanced etc, will be provided
+  as examples rather than supported layers with the intention to help users implement similar
+  functionality for themselves.
 
 ## Rules for Naming Props, Attributes, Uniforms and updateTriggers
 
-Summary: Custom layer props usually refer to either attributes or uniforms.
+Summary: Custom layer props usually corresponds to either attributes or uniforms
+of the underlying shaders.
 An accessor (like "getColor") always sets a WebGL attribute (whether a value
 or a function), and a normal prop (like "color") always sets a uniform
```

Comment:
>  normal prop (like "color") always sets a uniform

This case is not true, e.g. `fp64` flag (there are other examples). So I'd say "usually" rather than "always" in the uniform case.